doc/source/whatsnew/v1.0.0.rst

```diff
@@ -45,6 +45,7 @@ Backwards incompatible API changes
 Other API changes
 ^^^^^^^^^^^^^^^^^
 
+- :meth:`infer_type` will now return "integer-na" for integer and np.nan mix (:issue:`27283`)
```

Comment:
I think this needs to be ``:meth:`pandas.api.types.infer_dtype` ``

Reply:
Updated